app/controllers/api/automation_requests_controller.rb

```diff
@@ -41,5 +41,16 @@ def deny_resource(type, id, data)
     rescue => err
       action_result(false, err.to_s)
     end
+
+    def find_automation_requests(id)
+      klass = collection_class(:requests)
+      return klass.find(id) if User.current_user.admin_user?
+      klass.find_by!(:requester => User.current_user, :id => id)
+    end
+
+    def automation_requests_search_conditions
```

Comment:
Same as above - this issue is broader than this PR, so might be best done in a follow up